builders/ethereum/dev-env/thirdweb.md

```diff
@@ -110,184 +112,339 @@ thirdweb offers SDKs for a range of programming languages, such as React, React
-
-    function App() {
-      const { contract } = useContract(contractAddress);
-      const { mutateAsync, isLoading, error } = useContractWrite(
-        contract,
-        'INSERT_NAME'
-      );
-
-      return (
-        <Web3Button
-          contractAddress={contractAddress}
-          // Calls the 'INSERT_NAME' function on your smart contract
-          // with 'INSERT_ARGUMENT' as the first argument
-          action={() => mutateAsync({ args: ['INSERT_ARGUMENT'] })}
-        >
-          Send Transaction
-        </Web3Button>
-      );
-    }
-    ```
+```typescript title="Reading Contract State"
+import { readContract } from 'thirdweb';
 
-### Connect Wallet {: #connect-wallet }
+const balance = await readContract({
+  contract: contract,
+  method: 'function balanceOf(address) view returns (uint256)',
+  params: ['0x123...'],
+});
+```
 
-There are a couple of ways that you can create a custom [connect wallet](https://portal.thirdweb.com/react/connecting-wallets){target=\_blank} experience. You can use the [`ConnectWallet` component](https://portal.thirdweb.com/react/connecting-wallets#using-the-connect-wallet-button){target=\_blank} or, for a more customizable approach, you can use the [`useConnect` hook](https://portal.thirdweb.com/react/connecting-wallets#using-hooks){target=\_blank}.
+For a function that takes no parameters, such as the number function that returns the current number stored in the [incrementer contract](https://moonbase.moonscan.io/address/0xa72f549a1a12b9b49f30a7f3aeb1f4e96389c5d8){target=\_blank}, you simply need to provide the function name as follows: 
 
-The following example will show you how to use the `ConnectWallet` component. To go this route, you will need to specify the supported wallets and pass them to your provider.
+```typescript title="Reading Contract State"
+import { readContract } from 'thirdweb';
 
-```javascript
-import {
-  ThirdwebProvider,
-  metamaskWallet,
-  coinbaseWallet,
-  walletConnectV1,
-  walletConnect,
-  safeWallet,
-  paperWallet,
-} from '@thirdweb-dev/react';
+const number = await readContract({
+  contract: contract,
+  method: 'number',
+  params: [],
+});
+```
 
-function MyApp() {
-  return (
-    <ThirdwebProvider
-      supportedWallets={[
-        metamaskWallet(),
-        coinbaseWallet(),
-        walletConnect({
-          projectId: 'INSERT_YOUR_PROJECT_ID', // optional
-        }),
-        walletConnectV1(),
-        safeWallet(),
-        paperWallet({
-          clientId: 'INSERT_YOUR_CLIENT_ID', // required
-        }),
-      ]}
-      activeChain={Moonbeam}
-    >
-      <App />
-    </ThirdwebProvider>
-  );
-}
+Did you know? With the [thirdweb CLI](https://portal.thirdweb.com/cli){target=\_blank}, you can easily and generate functions for all of the possible calls to a contract. To do so, run the following command in the command line: 
+
+```bash
+npx thirdweb generate INSERT-CHAIN-ID/INSERT-CONTRACT-ADDRESS
```

Comment:
INSERT_CHAIN_ID - use underscores

also not sure what this syntax is? is it one or the other? or is the format actually chain-id/contract?

Reply:
Both, added additional context